### PR TITLE
Adding web::block to all routes.

### LINKED ISSUE #606: Improve performance by making database queries async / figure out better postgres caching.
This is not important now because CPU load is very low on the server, but something to keep in mind for the future.

>Make your `Peform` trait async (look into the async-trait crate), and put `web::block(move || {})` around each of your r2d2 accesses.
>
>It's a little unfortunate that y'all pulled your r2d2 .get()s out to the top-level of each route. That's definitely going to be a bottleneck for simultaneous requests, and also putting something _synchronous_ at that level basic removes the opportunity to do asynchronous operations inside the routes.
>
>Maybe instead of guarding the route with a pool.get(), you should just pass the pool into the route so you can .get() when you need it

### PR DESCRIPTION
**Pull request by dessalines**
Fixes #606

### CODE REVIEW

**Comment by dessalines** [comment]
@asonix Unfortunately I still don't know how to make the websocket server.rs properly non-blocking on the db calls...

Maybe [here](https://github.com/LemmyNet/lemmy/blob/master/server/src/websocket/server.rs#L899) or [here]([here](https://github.com/LemmyNet/lemmy/blob/master/server/src/websocket/server.rs#L971)?

**Comment by asonix** [comment]
If you're just putting web::block around oper.perform in the http route, you can do the same thing in the websocket file

File: server/src/websocket/server.rs [hunk 1]
```diff
@@ -896,10 +896,18 @@ where
...
+    actix_web::web::block(move || {
+      let parsed_data: Data = serde_json::from_str(&data)?;
+      let res = Oper::new(parsed_data).perform(pool, Some(ws_info))?;
+      to_json_string(&op, &res)
+    })
+    .await
+    .map_err(|e| match e {
+      actix_web::error::BlockingError::Error(e) => e,
+      _ => APIError::err("Operation canceled").into(),
+    })
```
Comment (dessalines) [inline server/src/websocket/server.rs:910]: @asonix This compiles, but I'm not totally sure if its correct. I'm gonna try to import one of my production DBs and see if it still seems like long DB operations are blocking others.

Comment (dessalines) [reply server/src/websocket/server.rs:910]: NM, looks like it works!

**Pull request merged**